src/graph/table.rs

```diff
@@ -103,52 +117,67 @@ pub(crate) fn group_event_records(
     event_records
 }
 
-fn varlociraptor_info(records: &[Record]) -> VarlociraptorInfo {
+fn event_names_to_prob_names(event_names: &Vec<String>) -> Vec<String> {
+    event_names.into_iter()
+        .map(|event_name| event_name.to_uppercase())
+        .map(|mut e_n| {
+            e_n.insert_str(0, "PROB_");
+            e_n
+        })
+        .collect()
+}
+
+fn get_value_from_prob_name(prob_name: &String, record: &Record) -> f32 {
+    let key = prob_name.parse::<Key>().unwrap();
+    record.info()
+        .get(&key)
+        .map(|f| match f.value() {
+            Some(Value::Float(f)) => *f,
+            _ => panic!("Expected float value for {}", prob_name),
+        })
+        .unwrap_or_else(|| panic!("'{}' info field not found", prob_name))
+}
+
+fn varlociraptor_info(records: &[Record], prob_info_fields: &Vec<String>) -> VarlociraptorInfo {
     let r = &records[0];
 
-    let [prob_present, prob_absent, prob_artifact] = [
-        ("PROB_PRESENT", "PROB_PRESENT".parse::<Key>().unwrap()),
-        ("PROB_ABSENT", "PROB_ABSENT".parse::<Key>().unwrap()),
-        ("PROB_ARTIFACT", "PROB_ARTIFACT".parse::<Key>().unwrap()),
-    ]
-    .map(move |(name, key)| {
-        r.info()
-            .get(&key)
-            .map(|f| match f.value() {
-                Some(Value::Float(f)) => *f,
-                _ => panic!("Expected float value for {}", name),
-            })
-            .unwrap_or_else(|| panic!("'{}' info field not found", name))
-    });
+    let prob_absent = get_value_from_prob_name(&String::from("PROB_ABSENT"), r);
+    let prob_artifact = get_value_from_prob_name(&String::from("PROB_ARTIFACT"), r);
+    let event_probs: HashMap<String, f32> = prob_info_fields
+    .into_iter()
+    .map(|field_name| {
+        (field_name.clone(), get_value_from_prob_name(field_name, r))
+    }).collect();
     let af_key = "AF".parse::<vcf::header::format::Key>().unwrap();
     let allele_frequencies = r
         .genotypes()
         .iter()
         .map(|f| f[&af_key].value())
         .collect_vec();
     VarlociraptorInfo {
-        prob_present,
         prob_absent,
         prob_artifact,
+        event_probs,
         af_nanopore: allele_frequencies[0].map(|v| match v {
             vcf::record::genotype::field::Value::Float(f) => *f,
             _ => panic!("Expected float value for AF"),
         }),
     }
 }
 
-#[derive(Serialize, Deserialize, Debug, Clone, Copy)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub(crate) struct VarlociraptorInfo {
-    prob_present: f32,
     prob_absent: f32,
     prob_artifact: f32,
+    event_probs: HashMap<String, f32>,
     af_nanopore: Option<f32>,
```

Comment:
This is hardcoded, but basically defined by the scenario / sample configuration, perhaps we can determine this dynamically as well?

Reply:
Well, in the snakemake workflow using this, this is determined by a mixture of the `scenario.yaml` and the `config.yaml`. The scenario determines which events exist and CAN be used. And the `config.yaml` entries under `filter: fdr-control: events: <joint_event_name>: []` determine which events ARE actually used in the `joint_event`. These don't have to be all the available events. Does this explanation make it clearer? Should I make the command-line argument explanations clearer to get this across?

Reply:
Sorry, the highlighting is a bit off here; my comment was with respect to the af_nanopore entry

Reply:
Ah, I see. But how would we want to report this, then? There can be any number of samples in the scenario and we can determine this dynamically. But how do we output this in the table and how do we process this in the workflow? Use some prefix and then put all the allele frequencies in the resulting report (like you do with the `prob_` prefix there)? One problem would be serializing the records though. I was thinking about this for the (sub-)event probs, and then decided to simply sum up their probs into one value to get around it.

Reply:
Yeah, it's not that straightforward, so let's postpone this to the future™